bcp47-orphans/tests/Data/BCP47/AesonSpec.hs

```diff
@@ -0,0 +1,13 @@
+{-# LANGUAGE TypeApplications #-}
+
+module Data.BCP47.AesonSpec (spec) where
+
+import Data.Aeson (decode, encode)
+import Data.BCP47 (BCP47)
+import Data.BCP47.Aeson ()
+import Test.Hspec
+import Test.QuickCheck (property)
+
+spec :: Spec
+spec = describe "ToJSON/FromJSON" . it "can roundtrip" . property $ \tag ->
```

Comment:
Just having some fun, not claiming this DRY is worth its own complexity:

```hs
-- tests/SpecHelper.hs
roundtrips
  :: (Eq a, Applicative f)
  => (a -> b) -- ^ encode
  -> (b -> f a) -- ^ decode
  -> Bool
roundtrips to from x = from (to x) == pure x

-- :trollface: 
roundtrips to from = (==) <$> from . to <*> pure

-- AesonSpec.hs
... property $ roundtrips @BCP47 encode decode

-- PathPieceSpec.hs
... property $ roundtrips @BCP47 toPathPiece fromPathPiece

-- And so on
```

Reply:
> `tests/SpecHelper.hs`

:thinking:

Reply:
Sorry was suggesting whatever centralized test prelude you want, `TestImport`? Anything's a cargo-cult anwyay...

- `SpecHelper` parrots rspec, which is good (IMO) because that's what `hspec` is; these are Specs, help them
- `TestImport` parrots Yesod, which is odd (IMO), but we've done that all over all of our packages already, so it's more our own standard now?
- `{Package}.Test.Prelude` is ideal (IMO) if thinking in a vacuum, but going with something completely new is always a Bad Idea, sadly